tests/integration/ha/helpers.py

```diff
@@ -300,8 +313,121 @@ async def send_control_signal(
         await ops_test.model.applications[app_name].add_unit(count=1)
         await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=1000)
 
-    cmd = f"run --unit {unit_name} -- pkill --signal {signal} -f {PROCESS}"
-    return_code, _, _ = await ops_test.juju(*cmd.split())
+    kill_cmd = f"exec --unit {unit_name} -- pkill --signal {signal} -f {PROCESS}"
+    return_code, stdout, stderr = await ops_test.juju(*kill_cmd.split())
 
     if return_code != 0:
-        raise Exception(f"Expected kill command {cmd} to succeed instead it failed: {return_code}")
+        raise Exception(
+            f"Expected kill command {kill_cmd} to succeed instead it failed: {return_code}, {stdout}, {stderr}"
+        )
+
+
+def get_password(model_full_name: str) -> str:
+    # getting relation data
+    show_unit = subprocess.check_output(
+        f"JUJU_MODEL={model_full_name} juju show-unit {APP_NAME}/0",
```

Comment:
**nitpick:** I know these were copied from the older (and worse) `tests/integration/helpers.py`, but could we pass `unit_name` as an argument to the function, and use it here instead of `*/0`?

Looking back, refactoring those things was always a pain.

Reply:
Is it needed in this case though? Result is not dependent on unit number.

Reply:
True, but if we refactor the test, we might have different unit numbers. 
Grabbing something like:

```python
ops_test.model.applications[APP_NAME].units[0].name
```

gives us more reliability for future refactors.